### PR TITLE
Add unit tests for JMockit Delegate to mockito migration. Also add test case for comments

### PR DESCRIPTION
**Pull request by shivanisky**
At the moment, JMockit Delegates are not migrated to mockito as mentioned in issue
- https://github.com/openrewrite/rewrite-testing-frameworks/issues/522. 

What I'm seeing is that they are being trashed with the template being printed out. These tests were written to try to replicate this issue, however I was unable to.
They may help anyone adding the feature for Delegate migration.

https://javadoc.io/doc/org.jmockit/jmockit/latest/mockit/Delegate.html

Also added one test case showing comments are not preserved.

I am wondering if it's better to have the test cases failing with the expected output and to disable them. Please let me know if that's preferred.

## Anyone you would like to review specifically?
@timtebeek @tinder-dthomson 

### Checklist
- [ ] I've added unit tests to cover both positive and negative cases
- [ ] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [ ] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Comment by timtebeek** [comment]
Thanks for the continued work on this! Indeed probably best to have the tests fail with the actual expected output, as opposed to documenting the status quo. No need to mark them `@Disabled` just yet, we can do that just before a merge if we want to delay any work towards this, and then have the tests reference an open issue with more information.

File: src/test/java/org/openrewrite/java/testing/jmockit/JMockitDelegateToMockitoTest.java
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/testing/jmockit/JMockitDelegateToMockitoTest.java:42]: I'm wondering what the right issue to tag these tests with is; would that be 
- https://github.com/openrewrite/rewrite-testing-frameworks/issues/325
```suggestion
    @Issue("https://github.com/openrewrite/rewrite-testing-frameworks/issues/325")
    void whenNoArgsVoidMethod() {
```

**Review by timtebeek — COMMENTED** [review]
Great to have these examples documented and runnable! One last question before we merge, but other than that good to go.

File: src/test/java/org/openrewrite/java/testing/jmockit/JMockitDelegateToMockitoTest.java
Comment (shivanisky) [reply src/test/java/org/openrewrite/java/testing/jmockit/JMockitDelegateToMockitoTest.java:42]: I am thinking this one: https://github.com/openrewrite/rewrite-testing-frameworks/issues/522 - have also mentioned it in PR description (just added it today). But ... 325 works well too - I'll let you decide.

Oh thanks - I wasn't sure how to tag an issue in this repo, in the past I've done it in a different way, thanks for showing it to me.

**Review by timtebeek — APPROVED** [review]

**Pull request merged**
